### ISSUE TITLE
Wrong code block renderer when documentation contains CLRF

### ISSUE THREAD

**Issue by CppCXY** [redhat-developer/lsp4ij#368]
in completion document:
![image](https://github.com/redhat-developer/lsp4ij/assets/40318218/bf4a72ac-c190-4024-9e01-389d9d357959)
in hover:
![image](https://github.com/redhat-developer/lsp4ij/assets/40318218/0c81c518-1cf9-4e2f-a679-2deaa1eb6910)

**Comment by angelozerr** [comment]
What si the result with JetBrains LSP support?

Could you share your LSP traces with hover please.

**Comment by CppCXY** [comment]
about this completion:
```log
[Trace - 19:27:54] Received response 'completionItem/resolve - (47)' in 3ms.
Result: {
  "label": "f",
  "labelDetails": {
    "detail": "(a)",
    "description": "nil"
  },
  "kind": 2,
  "documentation": {
    "kind": "markdown",
    "value": "```lua\nfunction f(a:aa) -\u003e nil\n```\n\n___\n```plaintext\nparams: a\r\n\n```\n\n___\n"
  },
  "data": "98784247821"
}
```

**Comment by CppCXY** [comment]
This is my own [Lua language server](https://github.com/CppCXY/EmmyLuaAnalyzer), not Sumneko Lua. I have not implemented a JetBrains LSP version for it.

**Comment by angelozerr** [comment]
According [this comment](https://github.com/redhat-developer/lsp4ij/pull/370#issuecomment-2178827960)  @fbricon I think we should keep like this, right?

**Comment by CppCXY** [comment]
another question about the render.
the return of the protocol is like this:
```log
[Trace - 15:01:34] Received response 'textDocument/hover - (9)' in 3ms.
Result: {
  "contents": {
    "kind": "markdown",
    "value": "```lua\nfunction insert(\r\n    list:table,\n    pos:number,\n    value:any\n)\n-\u003e number\n```\n\nin class `table`\r\nInserts element `value` at position `pos` in `list`, shifting up the\n elements to `list[pos]`, `list[pos+1]`, `···`, `list[#list]`. The default\n value for `pos` is ``#list+1`, so that a call `table.insert(t,x)`` inserts\n `x` at the end of list `t`.\r\nGo to: [table](file:///C:/Users/zc/.vscode/extensions/tangzx.emmylua-0.7.7-win32-x64/server/win32-x64/win32-x64/Resources/std/builtin.lua#91:4-91:15)|[number](file:///C:/Users/zc/.vscode/extensions/tangzx.emmylua-0.7.7-win32-x64/server/win32-x64/win32-x64/Resources/std/builtin.lua#40:4-40:16)|[any](file:///C:/Users/zc/.vscode/extensions/tangzx.emmylua-0.7.7-win32-x64/server/win32-x64/win32-x64/Resources/std/builtin.lua#95:4-95:13)\r\n"
  }
}
```

but I see: 
![image](https://github.com/redhat-developer/lsp4ij/assets/40318218/6605e294-eb22-4862-b521-dc85449cde5c)

the error in color might be my fault, but why has "number" become "numbe"?

in vscode:
![image](https://github.com/redhat-developer/lsp4ij/assets/40318218/d6579d6b-04ea-4cd6-8e51-3b43df5140f9)

**Comment by fbricon** [comment]
@angelozerr looks like an off-by-1 error in the code block renderer. Hopefully it should be easy to reproduce with a unit test

**Comment by deanmaster** [comment]
Hello everyone,

Probably this is not the right place, but can you show me documentation to adjust the clientCapabilities in this plugin or all clientCapabilities are hardcoded not matter what are supported from server.

Thanks,
Tuan Do

**Comment by angelozerr** [comment]
@deanmaster could you create a New issue please. 

It seems it is an another topic.

To give you a quick answer, client capabilities are hard coded because it defines the LSP4IJ capabilities.

In lsp client initlialize the lsp server by setting the client capabilitues. In vscode context, vscode use the hard coded vscode client capabilities. LSP4IJ does the same thing.

After this initialize, server send to client the server capabilities.

In other words you must not customize the client capabilities or perhaps I miss something?

**Comment by angelozerr** [comment]
> @angelozerr looks like an off-by-1 error in the code block renderer. Hopefully it should be easy to reproduce with a unit test

I need to reproduce it, but I wonder why we have not this problem with other ls?

@CppCXY have you every time this problem?

**Comment by CppCXY** [comment]
> > @angelozerr looks like an off-by-1 error in the code block renderer. Hopefully it should be easy to reproduce with a unit test
> 
> I need to reproduce it, but I wonder why we have not this problem with other ls?
> 
> @CppCXY have you every time this problem?

always meet this problem:
![image](https://github.com/redhat-developer/lsp4ij/assets/40318218/820e3bff-e676-4b2e-affa-68339a7cdeaf)

**Comment by angelozerr** [comment]
@deanmaster you dont need to build yourself the plugin. See 'Testing nightly builds' section from the README.md

**Comment by deanmaster** [comment]
@angelozerr  perfect, I just install nightly i can see it's working great!

**Comment by angelozerr** [comment]
> @angelozerr perfect, I just install nightly i can see it's working great!

Great, but we need to fix some existing issues.

**Comment by CppCXY** [comment]
> > > @angelozerr looks like an off-by-1 error in the code block renderer. Hopefully it should be easy to reproduce with a unit test
> > 
> > 
> > I need to reproduce it, but I wonder why we have not this problem with other ls?
> > @CppCXY have you every time this problem?
> 
> always meet this problem

The reason for this issue should be that I occasionally used CRLF in hover.

**Comment by angelozerr** [comment]
@CppCXY first I have renamed your issue.

I hav etried to do a quick fix.

Could you please try to install the LSP4IJ zip at https://github.com/redhat-developer/lsp4ij/actions/runs/10141756236

![image](https://github.com/user-attachments/assets/85c8f713-dafe-46dd-a9b1-8f54a0e05712)

and tell me if it fixes your issue please.

Thanks.

**Comment by CppCXY** [comment]
> and tell me if it fixes your issue please.
> 
> Thanks.

I tested it with the previous version, and it is indeed fixed.

**Comment by angelozerr** [comment]
> I tested it with the previous version, and it is indeed fixed.

That's great! Thanks for your feedback.

**Comment by angelozerr** [comment]
@CppCXY I have published nightly build with the fix.
